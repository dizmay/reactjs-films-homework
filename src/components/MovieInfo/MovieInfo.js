import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import styles from './MovieInfo.scss';

const MovieInfo = ({
  title,
  rate,
  genres,
  black,
  white,
}) => {
  const titleStyle = classNames(
    styles.info__title,
    {
      [styles.info__title_black]: black,
      [styles.info__title_white]: white,
    },
  );
  return (
    <div className={styles.info__container}>
      <div className={styles.info}>
        <span className={titleStyle}>{ title }</span>
        <div className={styles.info__rate}>{ rate.toFixed(1) }</div>
      </div>
      <div className={styles.info__genres}>{ genres }</div>
    </div>
  );
};

MovieInfo.defaultProps = {
  title: '',
  rate: 0,
  genres: '',
  black: true,
  white: false,
};

MovieInfo.propTypes = {
  title: PropTypes.string,
  rate: PropTypes.number,
  genres: PropTypes.string,
  black: PropTypes.bool,
  white: PropTypes.bool,
};

export default MovieInfo;
