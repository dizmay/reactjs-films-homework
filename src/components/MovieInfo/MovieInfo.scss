* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

.info__container {
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.info {
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: space-between;
}

.info__title {
  text-transform: uppercase;
  font-size: 1.25rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.info__title_black {
  color: #000;
}

.info__title_white {
  color: #fff;
}

.info__rate {
  font-size: 1rem;
  font-weight: bold;
  color: #00a3f5;
  border: 0.15rem solid #00a3f5;
  border-radius: 0.2rem;
  padding: 0.2rem;
  margin-left: 0.5rem;
}

.info__genres {
  width: 100%;
  color: #00a3f5;
}
