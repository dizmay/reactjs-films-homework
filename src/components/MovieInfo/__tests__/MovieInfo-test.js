import React from 'react';
import renderer from 'react-test-renderer';
import MovieInfo from '../MovieInfo';

it('renders correctly', () => {
  const tree = renderer.create(<MovieInfo />).toJSON();
  expect(tree).toMatchSnapshot();
});
