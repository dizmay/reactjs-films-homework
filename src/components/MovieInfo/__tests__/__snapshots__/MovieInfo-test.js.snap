// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="info__container"
>
  <div
    className="info"
  >
    <span
      className="info__title info__title_black"
    >
      
    </span>
    <div
      className="info__rate"
    >
      0.0
    </div>
  </div>
  <div
    className="info__genres"
  >
    
  </div>
</div>
`;
